connectd/tor.c

```diff
@@ -59,11 +59,12 @@ static struct io_plan *connect_finish(struct io_conn *conn,
 
 	if ( connect->buffer[1] == '\0') {
 		if ( connect->buffer[3] == SOCKS_TYP_IPV6) {
+			/* Read rest of response */
 			return io_read(conn,
-				       (connect->buffer + SIZE_OF_RESPONSE -
-					SIZE_OF_IPV4_RESPONSE),
+				       connect->buffer + SIZE_OF_RESPONSE +
+				       SIZE_OF_IPV4_RESPONSE,
 				       SIZE_OF_IPV6_RESPONSE -
-				       SIZE_OF_RESPONSE - SIZE_OF_IPV4_RESPONSE,
+				       SIZE_OF_IPV4_RESPONSE,
 				       &connect_finish2, connect);
 
 		} else if ( connect->buffer[3] == SOCKS_TYP_IPV4) {
```

Comment:
tACK  09caab5eef76b6599c262f565b15ece6b07378d

It's not easy to test since even if you call on ipv6 the Tor circuit will almost always return an empty IP4 return code.
Since we call on DOMAIN and never use the empty ipv4/6 return anyway.

As ipv6 often leaks mac, ipv6 and Tor are not the best match, but anyway   
Luckily the buffer[-2] went in the void

Good Catch!  :+1::